shared/js/views.jsx

```diff
@@ -831,6 +881,127 @@ loop.shared.views = (function(_, mozL10n) {
+      loop.store.StoreMixin("remoteCursorStore")
+    ],
+
+    propTypes: {
+      videoElementSize: React.PropTypes.object
+    },
+
+    getInitialState: function() {
+      return {
+        realVideoSize: null,
+        videoLetterboxing: null
+      };
+    },
+
+    componentWillMount: function() {
+      if (!this.state.realVideoSize) {
+        return;
+      }
+
+      this._calculateVideoLetterboxing();
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      // In this case link generator or link clicker have resized their windows 
+      // so we need to recalculate the video letterboxing
+      this._calculateVideoLetterboxing(this.state.realVideoSize, nextProps.videoElementSize);
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      if (!this.state.realVideoSize || !nextState.realVideoSize) {
+        return;
+      }
+
+      if (!this.state.videoLetterboxing) {
+        // If this is the first time we receive the event, we must calculate the
+        // video letterboxing
+        this._calculateVideoLetterboxing();
+        return;
+      }
+
+      if (nextState.realVideoSize.width !== this.state.realVideoSize.width ||
+        nextState.realVideoSize.height !== this.state.realVideoSize.height) {
+        // TODO: Update ratios because of the change of the stream dimension
+
+        // In this case link generator has resized his window so we need to
+        // recalculate the video letterboxing
+        this._calculateVideoLetterboxing(nextState.realVideoSize);
+      }
+    },
+
+    _calculateVideoLetterboxing: function(realVideoSize, videoElementSize) {
+      realVideoSize = realVideoSize || this.state.realVideoSize;
+      videoElementSize = videoElementSize || this.props.videoElementSize;
+      var streamVideoWidth = 0;
+      var streamVideoHeight = 0;
+
+      var clientWidth = videoElementSize.clientWidth;
+      var clientHeight = videoElementSize.clientHeight;
+
+      var realVideoWidth = realVideoSize.width;
+      var realVideoHeight = realVideoSize.height;
+
+      if (realVideoWidth >= clientWidth) {
+        // Reduce video width
+        streamVideoWidth = clientWidth;
+        streamVideoHeight = (realVideoHeight * clientWidth) / realVideoWidth;
+
+        // In this case we need to do extra work because the video can't be 100% width
+        if (streamVideoHeight > clientHeight) {
+          streamVideoWidth = (realVideoWidth * clientHeight) / realVideoHeight;
+          streamVideoHeight = clientHeight;
+        }
+      } else if (realVideoHeight >= clientHeight) {
+        streamVideoWidth = (realVideoWidth * clientHeight) / realVideoHeight;
+        streamVideoHeight = clientHeight;
+      } else {
+        // In this case we should keep the real dimension of the video since the
+        // video fits in the screen without reducing width or height
+        streamVideoWidth = realVideoWidth;
+        streamVideoHeight = realVideoHeight;
+      }
```

Comment:
This last else has the wrong assumption of what happens for small stream video sizes. It actually stretches to fit the available width or height instead of keeping the original size.

All of the conditionals can be simplified by comparing width/height ratios of the element and stream. I've committed this here: https://github.com/Mardak/loop/commit/23cbc16940b1ef0b1c038376525783128c2913cd and you can just integrate it into your commit.